mapstructure_test.go

```diff
@@ -2334,7 +2389,26 @@ func TestDecode_StructTaggedWithOmitempty_KeepNonEmptyValues(t *testing.T) {
 		"visible-map":      emptyMap,
 		"omittable-map":    map[string]interface{}{"k": "v"},
 		"visible-nested":   emptyNested,
-		"omittable-nested": &Nested{},
+		"omittable-nested": map[string]interface{}{
+			"Vbar": map[string]interface{}{
+				"Vbool":       false,
+				"Vdata":       interface{}(nil),
+				"Vextra":      "",
+				"Vfloat":      float64(0),
+				"Vint":        0,
+				"Vint16":      int16(0),
+				"Vint32":      int32(0),
+				"Vint64":      int64(0),
+				"Vint8":       int8(0),
+				"VjsonFloat":  float64(0),
+				"VjsonInt":    0,
+				"VjsonNumber": json.Number(""),
+				"VjsonUint":   uint(0),
+				"Vstring":     "",
+				"Vuint":       uint(0),
+			},
+			"Vfoo": "",
+		},
```

Comment:
I think this test change was masking the bug described in #231 

This PR should not have changed any tests with omitempty, right?

Reply:
Yes you're probably right. I thought this would be safe because this only added tests but I see this is a change.